### PR TITLE
feat: BBS+ signer

### PR DESCRIPTION
**Pull request by kdimak**
This PR provides an implementation of BBS+ signer. It also provides a generator of public/private key pair. We can now generate a key pair, sing one or more messages, and verify the created signature (this functionality was already present).

Signed-off-by: Dmitriy Kinoshenko <dkinoshenko@gmail.com>

Relates to #2222 . To close this issue, in the follow-up PR we will use the BBS+ signer for VC.

### CODE REVIEW

File: pkg/doc/bbs/bbs12381g2pub/bbs.go [hunk 1]
```diff
@@ -83,7 +88,61 @@ func (b BBSG2Pub) Verify(messages [][]byte, sigBytes, pubKeyBytes []byte) error
...
+
+// SignWithKeyPair signs the one or more messages using BBS+ key pair.
+func (bbs *BBSG2Pub) SignWithKeyPair(messages [][]byte, privKey *PrivateKey, pubKey *PublicKey) ([]byte, error) {
+	var err error
+
+	messagesFr := make([]*SignatureMessage, len(messages))
+	for i := range messages {
+		messagesFr[i], err = ParseSignatureMessage(messages[i])
+		if err != nil {
+			return nil, fmt.Errorf("parse signature message: %w", err)
```
Comment (Baha-sk) [inline pkg/doc/bbs/bbs12381g2pub/bbs.go:113]: add the index `i` in the error message to help troubleshoot errors

File: pkg/doc/bbs/bbs12381g2pub/bbs.go [hunk 2]
```diff
@@ -83,7 +88,61 @@ func (b BBSG2Pub) Verify(messages [][]byte, sigBytes, pubKeyBytes []byte) error
...
+
+	if len(messages) == 0 {
+		return nil, errors.New("messages are not defined")
+	}
+
+	return bbs.SignWithKeyPair(messages, privKey, privKey.PublicKey())
+}
+
+// SignWithKeyPair signs the one or more messages using BBS+ key pair.
+func (bbs *BBSG2Pub) SignWithKeyPair(messages [][]byte, privKey *PrivateKey, pubKey *PublicKey) ([]byte, error) {
```
Comment (Baha-sk) [inline pkg/doc/bbs/bbs12381g2pub/bbs.go:106]: doesn't `privKey` include the public key as well ? can't we just pass privKey alone then do privKey.PublicKey() inside the function to fetch its public key?

what happens if someone passes a public key not matching the private one? I think its unnecessary to pass pubKey to this function.

Comment (kdimak) [reply pkg/doc/bbs/bbs12381g2pub/bbs.go:106]: I agree with this point. No need to pass a `pubKey` as it's created from `privKey` by a straightforward algorithm.

File: pkg/doc/bbs/bbs12381g2pub/bbs.go (hunk 1 above)
Comment (kdimak) [reply pkg/doc/bbs/bbs12381g2pub/bbs.go]: Improved.

File: pkg/doc/bbs/bbs12381g2pub/bbs.go (hunk 2 above)
Comment (kdimak) [reply pkg/doc/bbs/bbs12381g2pub/bbs.go]: Corrected the API.

File: pkg/doc/bbs/api.go [hunk 3]
```diff
@@ -9,8 +9,15 @@ package bbs
...
 type BBS interface {
 
-	// Verify will verify each signature against a public key
+	// Verify will verify an aggregated signature of one or more messages against a public key
 	// returns:
 	// 		error in case of errors or nil if signature verification was successful
 	Verify(messages [][]byte, signature, pubKey []byte) error
+
+	// Sign will sign create signature of each message and aggregate it into a single signature using
+	// a matching signature primitive in kh key handle.
```
Comment (llorllale) [inline pkg/doc/bbs/api.go:18]: `privKey` is a raw key, not a handle

**Review by Baha-sk — APPROVED** [review]

**Review by fqutishat — APPROVED** [review]

File: pkg/doc/bbs/api.go (hunk 3 above)
Comment (kdimak) [reply pkg/doc/bbs/api.go]: Corrected.

**Review by llorllale — APPROVED** [review]

**Pull request merged**